src/haddock/gear/prepare_run.py

```diff
@@ -315,6 +315,12 @@ def setup_run(
             dec_all=True,
         )
 
+    first_module_id = list(modules_params.keys())[0]
+    if (topoaa_module_id := "topoaa.1") in modules_params.keys():
+        topology_params = modules_params[topoaa_module_id]
+    else:
+        topology_params = {}
```

Comment:
I see in the old version this `topoaa.1` was also present, can you explain why?

Reply:
Required by the function `populate_topology_molecule_params()`, to map the `molX` parameters info to input molecules, in the case `topoaa` is the first module.
Without this, we loose the information.

Reply:
Added comments on the code directly